lib/client/ClientRenderingWorkflow.js

```diff
@@ -16,153 +16,145 @@ var environmentConfigForAllRequests;
 
 var ClientRenderingWorkflow = {
 
-    createHandlerFrom: function(originalHandler) {
-        return function() {
-            currentRequestId = currentRequestId + 1;
-
-            var clientRequest = ClientRequest.from(window, currentRequestId, environmentConfigForAllRequests);
-            var clientResponse = ClientResponse.from(window);
-
-            var router = this;
-            var LayoutForRoute = router.layout;
-            var errorViewMappingForRoute = router.errorViewMapping;
-            var onRouteStart = router.onRouteStart;
-            var onRouteComplete = router.onRouteComplete;
-
-            var isFirstPageLoad = !currentLayout;
-            var layoutForRouteIsSameAsCurrent = isFirstPageLoad || currentLayout.isSameTypeAs(LayoutForRoute);
-            var shouldUseCurrentLayout = !isFirstPageLoad && layoutForRouteIsSameAsCurrent;
-            var layout = shouldUseCurrentLayout ? currentLayout : new LayoutForRoute();
-            var layoutDelegate = LayoutDelegate.from(layout, LayoutForRoute);
-            var routeArguments = new Array(arguments.length);
-
-            for (var i = 0; i < routeArguments.length; i++) {
-                routeArguments[i] = arguments[i];
-            }
+    execute: function(router, originalHandler, params) {
```

Comment:
the only things changed in this file:

- function signature
- function returns promise of client rendering rather than a function
- calculating routeArguments is replaced with params parameter